src/litdata/processing/data_processor.py

```diff
@@ -1119,17 +1137,13 @@ def run(self, data_recipe: DataRecipe) -> None:
 
             # Exit early if all the workers are done.
             # This means there were some kinda of errors.
+            # TODO: Check whether this is still required.
```

Comment:
I don't know if we still need this. All It does is give random test fails in CI (when optimize function is very small).

https://github.com/Lightning-AI/litdata/actions/runs/10898567830/job/30249673425?pr=348